doc/magic_design_doc.md

```diff
@@ -0,0 +1,77 @@
+# Magic Command Design Doc
+
+## Overview
+
+Magic command wraps a pysqlflow client. It pretty prints the logs and tables.
```

Comment:
magic command is not just for pretty prints. It should be able to assign data to a variable and use it for charting or calculation.

Reply:
> It should be able to assign data to a variable and use it for charting or calculation.

Sure Python programmers would benefit from this feature. However, I am worrying about this pythonic user experience would be so different from SQL experience.

Reply:
notebook display variable with format, for example, it displays data frame as table. Need find more detail about display variable in notebook